dm/syncer/error.go

```diff
@@ -211,13 +212,66 @@ func (s *Syncer) handleSpecialDDLError(tctx *tcontext.Context, err error, ddls [
 	}
 	// TODO: add support for downstream alter pk without schema
 
+	// it handles the operations of DDL when encountering `invalid connection`
+	invalidConnF2 := func(tctx *tcontext.Context, err error, ddls []string, index int, conn *dbconn.DBConn, createTime int64) error {
+		if len(ddls) == 0 || index > len(ddls)-1 || errors.Cause(err) != mysql.ErrInvalidConn || createTime == -1 {
+			return err // return the original error
+		}
+
+		statusChan := make(chan string)
+		var status string
+		var err2 error
+
+		for {
+			status, err2 = getDDLStatusFromTiDB(tctx, conn, ddls[index], createTime) // status is synced
```

Comment:
```suggestion
			status, err2 := getDDLStatusFromTiDB(tctx, conn, ddls[index], createTime) // status is synced
```

no need to declare them outside the loop